modules/local/ensembl_genome_download.nf

```diff
@@ -0,0 +1,55 @@
+// Module that downloads all necessary genome files from Ensembl.
+// The module checks that the MD5 checksums match before releasing the data.
+// It also uncompresses the files, since we want bgzip compression.
+process ENSEMBL_GENOME_DOWNLOAD {
+    tag "$assembly_accession"
+    label 'process_single'
+
+    conda (params.enable_conda ? "bioconda::wget=1.18" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/gnu-wget:1.18--h7132678_6' :
+        'quay.io/biocontainers/gnu-wget:1.18--h7132678_6' }"
+
+    input:
+    tuple val(analysis_dir), val(ensembl_species_name), val(assembly_accession)
+
+    output:
+    tuple val(meta), path("*.fa") , emit: fasta
+    path  "versions.yml"          , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+
+    // e.g. https://ftp.ensembl.org/pub/rapid-release/species/Agriopis_aurantiaria/GCA_914767915.1/genome/Agriopis_aurantiaria-GCA_914767915.1-softmasked.fa.gz
+    def ftp_path = params.ftp_root + "/" + ensembl_species_name + "/" + assembly_accession + "/genome"
+    def remote_filename_stem = ensembl_species_name + "-" + assembly_accession
+
+    meta = [
+        id : assembly_accession + ".masked.ensembl",
+        outdir: analysis_dir,
+        accession : assembly_accession
+    ]
```

Comment:
Same as https://github.com/sanger-tol/ensemblgenedownload/pull/1#discussion_r985801737

> Try to implement this in the `download` subworkflows. Suggested input:
> `tuple val(meta), val(ftp_path), val(remote_filename_stem)`

Reply:
Addressed the same way as in ensemblgenedownload. Those variables are built in the subworkflow and given to the module